### PR TITLE
Walmart scraper

### PR DESCRIPTION
**Pull request by pfurbacher**
Solution for #247

### CODE REVIEW

File: site-scrapers/Walmart/index.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+    const results = [];
+
+    const stores = fetchService.getStores();
+    if (loginSucceeded) {
+        Promise.all([
+            Object.keys(stores).forEach(async (storeNumber) => {
+                let response = await fetchService.fetchStoreAvailability(
+                    page,
+                    storeNumber
```
Comment (harcod) [inline site-scrapers/Walmart/index.js:57]: Should this be `await Promise.all(...` ? (I'm not a Promise expert)

Comment (harcod) [reply site-scrapers/Walmart/index.js:57]: My IDE flagged this during it's "code inspection"

Comment (pfurbacher) [reply site-scrapers/Walmart/index.js:57]: Not an expert either. Turns out that it's not needed, and cannot recall why I wrapped that code in the Promise.all. If I remove it, things work just fine.

**Comment by harcod** [comment]
I found a location with vaccine in Oklahoma.  The code isn't finding availability there even though there is.  

And the format of the returned response in Chrome inspection: (This is nothing like `slotDays` that `parseAvailability` is expecting)
```
{
  "status": "1",
  "message": "OK",
  "data": {
    "storeNbr": 576,
    "inventory": [
      {
        "shortName": "PFIZER VACC COVID-19 INJ",
        "productMdsFamId": 181947,
        "quantity": 1584
      },
      {
        "shortName": "MODERNA VAC COVID-19 INJ",
        "productMdsFamId": 181948,
        "quantity": 0
      },
      {
        "shortName": "JANSSEN VACCINE COVID-19 INJ",
        "productMdsFamId": 182366,
        "quantity": 0
      }
    ]
  }
}
```

I added this to my `config.js` to attempt to get this location.
```
    576: {
        street: "207 S Memorial Dr",
        city: "Tulsa",
        zip: "74129",
    },
```

**Comment by pfurbacher** [comment]
Thanks for digging that up (how did you know to look in OK? Maybe anywhere in the conservative South would do). I was working off notes Liv had given me as to the JSON path to the data, since I had not seen any availability here. I'll change it and run it with your store added.

**Comment by harcod** [comment]
There is a subsequent call with a "Request Payload" of `{"startDate":"04222021","endDate":"04282021","imzStoreNumber":{"USStoreId":576}}` and this URL.
https://www.walmart.com/pharmacy/v2/clinical-services/time-slots/${accountId}

It returns your "slotDate" JSON information

**Comment by pfurbacher** [comment]
Yes, that's what the "slot-time" URL (with body) returns.

**Comment by pfurbacher** [comment]
I could be wrong, but from further investigation, trying to fetch using the time-slots URL with respective payload results in no response. So, to canvas all 48 stores, it looks like it will be necessary to use puppeteer to 

1. on the "Select a pharmacy in your area to confirm eligibility and schedule an appointment" page, set a zip code to catch 25 most easterly locations in MA
2. select each of the 25 locations offered per zip code setting, screen them for MA address (CT can also show up in the list),
3. remove the storeId from a list of all storeIds
4. repeat steps 1-3, moving the zip code west some 50 miles each time (manually figure these moves for least number of moves to catch all)

**Comment by livgust** [comment]
I'm going to merge this into the `walmart` branch I just made in our home repo so I can mess with it over the weekend (or anyone else can!).

**Comment by pfurbacher** [comment]
I'm pretty close but have been stuck on the fetch causing a Target closed exception. @Liv -- Can I update this PR with mostly working code and maybe you can figure out where I'm making a mistake?

**Pull request merged**